main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala

```diff
@@ -755,6 +755,17 @@ object ParsedAst {
       */
     case class RelationalChoose(sp1: SourcePosition, star: Boolean, exps: Seq[ParsedAst.Expression], rules: Seq[RelationalChoiceRule], sp2: SourcePosition) extends ParsedAst.Expression
 
+    /**
+      * Choose Expression.
+      *
+      * @param sp1   the position of the first character in the expression.
+      * @param star  whether this is a choose* expression.
+      * @param exp   the match expressions.
+      * @param rules the rules of the pattern match.
+      * @param sp2   the position of the last character in the expression.
+      */
+    case class Choose(sp1: SourcePosition, star: Boolean, exp: ParsedAst.Expression, rules: Seq[MatchRule], sp2: SourcePosition) extends ParsedAst.Expression
```

Comment:
Do we want to call it RestrictableChoose? Or perhaps move it above RelationalChoose?